modules/core/src/smithy4s/Service.scala

```diff
@@ -35,12 +35,71 @@ import kinds._
   *   metaprogramming.
   */
 trait Service[Alg[_[_, _, _, _, _]]] extends FunctorK5[Alg] with HasId {
+  /**
+   * A datatype (typically a sealed trait) that reifies an operation call within
+   * a service. It essentially captures the input and type indexes that the operation
+   * deals with.
+   *
+   * @tparam I: the input of the operation
+   * @tparam E: the error type associated to the operation (typically represented as a sealed-trait)
+   * @tparam O: the output of the operation
+   * @tparam SI: the streamed input of an operation. Operations can have unary components and streamed components.
+   *         For instance, an http call can send headers (unary I) and a stream of bytes (streamed SI) to the server.
+   * @tparam SO: the streamed output of the operation.
+   */
   type Operation[I, E, O, SI, SO]
+
+  /**
+   * An endpoint is the set of schemas tied to types associated with an operation, but
+   * also has a method to wrap the input in an operation instance :
+   * I => Operation[I, E, O, SI, SO]
+   */
   type Endpoint[I, E, O, SI, SO] = smithy4s.Endpoint[Operation, I, E, O, SI, SO]
+
+  /**
+   * This is a polymorphic function that runs an instance of an operation and produces an effect F.
+   */
   type Interpreter[F[_, _, _, _, _]] = PolyFunction5[Operation, F]
+
+  /**
+   * An interpreter specialised for effects of kind `* -> *`, like Try or monofunctor IO.
+   */
   type FunctorInterpreter[F[_]] = PolyFunction5[Operation, kinds.Kind1[F]#toKind5]
+
+  /**
+   * An interpreter specialised for effects of kind `* -> (*, *)`, like Either or bifunctor IO.
+   */
   type BiFunctorInterpreter[F[_, _]] = PolyFunction5[Operation, kinds.Kind2[F]#toKind5]
+
+  /**
+   * A polymorphic function that can take an Endpoint (associated to this service) and
+   * produces an handler for it, namely a function that takes the input type of the
+   * operation, and produces an effect.
+   */
+  type EndpointCompiler[F[_, _, _, _, _]] = PolyFunction5[Endpoint, kinds.Kind5[F]#handler]
+
+  /**
+   * A handler compiler specialised for effects of kind `* -> *`, like Try or monofunctor IO
+   */
+  type FunctorEndpointCompiler[F[_]] = PolyFunction5[Endpoint, kinds.Kind1[F]#handler]
+
+  /**
+   * A handler compiler specialised for effects of kind `* -> (*, *)`, like Either or bifunctor IO
+   */
+  type BiFunctorEndpointCompiler[F[_, _]] = PolyFunction5[Endpoint, kinds.Kind2[F]#handler]
+
+  /**
+   * A short-hand for algebras that are specialised for effects of kind `* -> *`.
+   *
+   * NB: this alias should be used in polymorphic implementations. When using the Smithy4s
+   * code generator, equivalent aliases that are named after the service are generated.
+   */
   type Impl[F[_]] = Alg[kinds.Kind1[F]#toKind5]
```

Comment:
nitpick: should this point to `FunctorAlgebra[Alg, F]` to make that type more discoverable?

Reply:
[0b5591a](https://github.com/disneystreaming/smithy4s/pull/908/commits/0b5591adaa6ffbaff1c0df45adad24178ac2e198)